src/compiler/utilities.ts

```diff
@@ -3294,11 +3294,21 @@ namespace ts {
         return !!name && isDynamicName(name);
     }
 
+    export function getComputedPropertyNameExpression(name: ComputedPropertyName): Expression {
+        return isElementAccessExpression(name.expression)
+                    ? skipParentheses(name.expression.argumentExpression)
+                    : name.expression;
+    }
+

```

Comment:
added a utility because after inlining the fix the ts-server died 5+ times,  and it turns out the same thing is needed elsewhere too, so figured i'd make it explicit.